### ISSUE TITLE
Block using a graphic effect other than brightness breaks graphic effect brightness, even if it is not active

### ISSUE THREAD

**Issue by jwzimmer** [LLK/scratch-flash#1294]
**TL;DR:** Brightness effect doesn't work as expected if another graphic effect block is in the editor, even if it doesn't become active when scripts are run.

Repro:
- Open project 152027812 or create a project with a script as shown in the screenshot
- Drag the "set x effect to y", or "change x effect by y" block into the editor, make x something other than brightness
- Press green flag
- Press red stop sign to re-set effects
- Change x to brightness
- Press green flag

<img width="246" alt="screen shot 2017-03-23 at 4 48 28 pm" src="https://cloud.githubusercontent.com/assets/1320456/24269901/6171d2f0-0fe9-11e7-93ad-16f60f348ac0.png">
<img width="158" alt="screen shot 2017-03-23 at 4 47 54 pm" src="https://cloud.githubusercontent.com/assets/1320456/24269900/616cb18a-0fe9-11e7-8753-808e5c281620.png">

![out](https://cloud.githubusercontent.com/assets/1320456/24269640/6f82447a-0fe8-11e7-860c-e1e8d83193fc.gif)

**Comment by TheLogFather** [comment]
Duplicate of https://github.com/LLK/scratch-flash/issues/570 ...?

I submitted a patch some time ago: https://github.com/LLK/scratch-flash/pull/972
(Could do with intensive testing on various hardware, though, since this AGAL stuff is tricky. Also, see discussion on above about 'special handling' of greys.)

**Comment by jwzimmer** [comment]
Tracking in https://github.com/LLK/scratch-flash/issues/1294.
